internal/server/main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+
+	"go_Playlist_gRPC/internal/playlist"
+	pb "go_Playlist_gRPC/internal/proto/music_player"
+)
+
+type server struct {
+	pb.UnimplementedMusicPlayerServer
+	Cmd chan playlist.Commands
+}
+
+func (s server) Play(ctx context.Context, e *pb.Empty) (*pb.PlaylistResponse, error) {
+	fmt.Println("Play")
+	s.Cmd <- playlist.Commands{Command: playlist.Play}
+	return &pb.PlaylistResponse{Code: 200, Message: "Play"}, nil
+}
+
+func (s server) Pause(ctx context.Context, e *pb.Empty) (*pb.PlaylistResponse, error) {
+	fmt.Println("Pause")
+	s.Cmd <- playlist.Commands{Command: playlist.Pause}
+	return &pb.PlaylistResponse{Code: 200, Message: "Pause"}, nil
+}
+
+func (s server) Next(ctx context.Context, e *pb.Empty) (*pb.PlaylistResponse, error) {
+	fmt.Println("Next")
+	s.Cmd <- playlist.Commands{Command: playlist.Next}
+	return &pb.PlaylistResponse{Code: 200, Message: "Next"}, nil
+}
+
+func (s server) Prev(ctx context.Context, e *pb.Empty) (*pb.PlaylistResponse, error) {
+	fmt.Println("Prev")
+	s.Cmd <- playlist.Commands{Command: playlist.Prev}
+	return &pb.PlaylistResponse{Code: 200, Message: "Prev"}, nil
+}
+
+func (s server) AddSong(ctx context.Context, req *pb.SongRequest) (*pb.PlaylistResponse, error) {
+	fmt.Printf("AddSong: name=%s, duration=%d\n", req.GetName(), req.GetDuration())
+	s.Cmd <- playlist.Commands{Command: playlist.AddSong, Song: playlist.Song{Name: req.GetName(), Duration: time.Duration(req.GetDuration())}}
+	return &pb.PlaylistResponse{Code: 200, Message: "AddSong"}, nil
+}
+
+func main() {
```

Comment:
Please consider unloading the `main` function and placing only global context and graceful shutdown using WaitGroups. For all other functionality, you can use wrappers.